opentracing/scope_managers/flask.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) The OpenTracing Authors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import threading
+
+from flask import _request_ctx_stack as stack
+
+from opentracing import Scope, ScopeManager
+from .constants import ACTIVE_ATTR
+
+
+class FlaskScopeManager(ScopeManager):
+    """
+    :class:`~opentracing.ScopeManager` implementation for **Flask**
+    that stores the :class:`~opentracing.Scope` in a Flask
+    :class:`RequestContext`, made accessible via
+    :attr:`flask._request_ctx_stack`.
+    """
+    def __init__(self):
+        self._tls = threading.local()
```

Comment:
Any reason not to inherit directly from `ThreadLocalScopeManager` and fallback to the base implementation when there's no `stack.top`?

Reply:
Are you suggesting this?
```
-class FlaskScopeManager(ScopeManager):
+class FlaskScopeManager(ThreadLocalScopeManager):
     """
     :class:`~opentracing.ScopeManager` implementation for **Flask**
     that stores the :class:`~opentracing.Scope` in a Flask
     :class:`RequestContext`, made accessible via
     :attr:`flask._request_ctx_stack`.
     """
-    def __init__(self):
-        self._tls = threading.local()
-
     @property
     def _context(self):
-        # Default to threading.local for usage outside
+        # Default to ThreadLocalScopeManager._tls_scope for usage outside
         # app/request context (unit tests)
         if stack.top is None:
-            return self._tls
+            return self._tls_scope
         return stack.top
```
Otherwise, if I switched `ThreadLocalScopeManager._tls_scope` to be the `_context` property that returns tls, `FlaskScopeManager` would only need to implement that.